src/main/ruby/truffleruby/core/exception.rb

```diff
@@ -469,6 +469,18 @@ def initialize(sig, message = undefined)
     @signo = signo
     super(name_with_prefix)
   end
+
+  private
+
+  def reached_top_level
+    begin
+      Signal.trap(@signo, 'SYSTEM_DEFAULT')
+    rescue ArgumentError
+      # some signals are reserved but we can raise them anyways
+    end
+    Truffle::POSIX.raise_signal(@signo)
```

Comment:
Is it important to use `raise(3)` here or would `Process.kill` works too?
`Process.kill` uses Java's `Signal.raise()` if possible and `kill(2)` otherwise, which might be nicer and avoids yet another way to raise signals.

Reply:
OK as-is from your general comment.